### PR TITLE
Praise be to DuARTe, #MATERIALYOLO

### PR DESCRIPTION
**Pull request by bryansills**
Bad androidpolice.com memes aside, this is the work I've been doing for the past few weeks. It brings Material Design to the app, even for God-forsaken phones still running 2.2.

NOTE: THIS IS NOT READY TO BE MERGED IN. I JUST WANTED TO GET THIS IN FRONT OF PEOPLE'S EYES BY THE START OF THIS WEEK.

Please checkout my branch, run the app on your phones, and tell me if I need to make any changes.

### CODE REVIEW

**Comment by HumbleBeeBumbleBee** [comment]
Tested in emulator:
* Nexus_One_API_7_arm ` minor crash on caret click in "Status" menu ` Android 2.1
http://ibin.co/24tWajaeovy4
http://ibin.co/24tWpPOT1gOX
return Caret on View scrobble cache works ...
* Nexus_4_API_10_intel_atom_x86 ` no crash ` Android 2.3.3
return Caret on "View scrobble cache" and "Status" works, API-10+ 
* Nexus_4_API_11_arm  ` no crash ` Android 3.0
* Nexus_4_API_14_arm ` no crash ` Android 4.0
* Nexus_4_API_15_intel_atom_x86 ` no crash ` Android 4.0.3
* Nexus_4_API_16_intel_atom_x86 ` no crash ` Android 4.1.2
!!!ONLY return Caret on "View scrobble cache" and "Status" works, API-16+ 
* Nexus_5_API_17_intel_atom_x86 ` no crash ` Android 4.2.2
* Nexus_5_API_18_intel_atom_x86 ` no crash ` Android 4.3.1
* Nexus_5_API_19_intel_atom_x86 ` no crash ` Android 4.4.2
* Nexus_5_API_21_intel_atom_x86 ` no crash ` Android 5.0.1
* Nexus_5_API_22_intel_atom_x86 ` no crash ` Android 5.0.1

**Comment by HumbleBeeBumbleBee** [comment]
Good meme. Guess I need to be into the code more to recognize a good meme when I'm shown one.

I guess there's a little problem with the back Caret all the way through. Maybe if it's fixed, back caret will work on all API SLS supports.

I will test on an actual phone in a little while.

**Comment by bryansills** [comment]
Fixed the crash on API 7. Fixed the back caret issues. I tested on API levels: 7, 10, 15, and 22. I also renamed some Activity files to replace the suffix "Screen" with "Activity". Please test scrobbling on a phone before merging; I had some issues with the `startService(Intent intent)` method, but I just want to confirm that other people are having issues before I start committing code kinda outside the scope of this pull request.

**Comment by HumbleBeeBumbleBee** [comment]
Tested in emulator:
* Nexus_One_API_7_arm ` no crash ` Android 2.1
All menus and back caret work on all API-7+
* Nexus_4_API_10_intel_atom_x86 ` no crash ` Android 2.3.3
* Nexus_4_API_11_arm  ` no crash ` Android 3.0 --
* Nexus_4_API_14_arm ` no crash ` Android 4.0 --
* Nexus_4_API_15_intel_atom_x86 ` no crash ` Android 4.0.3
* Nexus_4_API_16_intel_atom_x86 ` no crash ` Android 4.1.2
* Nexus_5_API_17_intel_atom_x86 ` no crash ` Android 4.2.2
* Nexus_5_API_18_intel_atom_x86 ` no crash ` Android 4.3.1 --
* Nexus_5_API_19_intel_atom_x86 ` no crash ` Android 4.4.2
* Nexus_5_API_21_intel_atom_x86 ` no crash ` Android 5.0.1 --
* Nexus_5_API_22_intel_atom_x86 ` no crash ` Android 5.1.1 --

I had one minor crash in API_18 (Android 4.3.1) in "Credentials" menu but I spent a good 5-10 minutes trying to recreate it, I could not, so I guess it was just the emulator being buggy. No worries.
On a bunch of the emulators (--) I noticed they freeze up occasionally. 

Everything looks good! I have to say, I like the Material Design features you've added. Very nice.
I'm not sure anybody else knows much about design specifications or how to implement them as well as you have.

My next post will describe any bugs I find in SLS after testing on my phones.

**Comment by HumbleBeeBumbleBee** [comment]
I am also wondering about the [compile size increase](http://ibin.co/251Ylh3d8UuD).
170 KB to 1,355 KB

My next post will describe any bugs I find in SLS after testing on my phones.

**Comment by HumbleBeeBumbleBee** [comment]
All the menus appear to be working. Screens work, etc.
However, I have one track in the cache and when select "Scrobble now" SLS crashes.

**Comment by bryansills** [comment]
Yup... that "Scrobble now" crash is what I'm talking about when I mentioned the `startService(Intent intent)` issues. I have something that fixes the issue, so I'll add it to the pull request and see what people have to say about the solution.

File: app/src/main/java/com/adam/aslfms/util/Util.java [hunk 1]
```diff
@@ -180,9 +180,10 @@ public static void warningDialog(Context ctx, String msg) {
 	public static void scrobbleIfPossible(Context ctx, NetApp napp,
 			int numInCache) {
 		if (numInCache > 0) {
-			Intent i = new Intent(ScrobblingService.ACTION_JUSTSCROBBLE);
-			i.putExtra("netapp", napp.getIntentExtraValue());
-			ctx.startService(i);
+            Intent intent = new Intent(ctx, ScrobblingService.class);
+			intent.setAction(ScrobblingService.ACTION_JUSTSCROBBLE);
+			intent.putExtra("netapp", napp.getIntentExtraValue());
+			ctx.startService(intent);
```
Comment (bryansills) [inline app/src/main/java/com/adam/aslfms/util/Util.java]: I'm guessing when bumping up the targetSDK from 12 to 22, something changed in how you can start up services. I was getting this crash:

```  java.lang.IllegalArgumentException: Service Intent must be explicit: Intent { act=com.adam.aslfms.service.justscrobble (has extras) }
            at android.app.ContextImpl.validateServiceIntent(ContextImpl.java:1680)
            at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1709)
            at android.app.ContextImpl.startService(ContextImpl.java:1693)
            at android.content.ContextWrapper.startService(ContextWrapper.java:516)
            at com.adam.aslfms.util.Util.scrobbleAllIfPossible(Util.java:196)
            at com.adam.aslfms.SettingsActivity.onPreferenceTreeClick(SettingsActivity.java:116)
            at android.preference.Preference.performClick(Preference.java:993)
            at android.preference.PreferenceScreen.onItemClick(PreferenceScreen.java:214)
            at android.widget.AdapterView.performItemClick(AdapterView.java:305)
            at android.widget.AbsListView.performItemClick(AbsListView.java:1146)
            at android.widget.AbsListView$PerformClick.run(AbsListView.java:3053)
            at android.widget.AbsListView$3.run(AbsListView.java:3860)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698) ```

wherever we called `startService(Intent intent)`. All the Service Intents get resolved by ScrobblingService, so I might as well make all the Intents explicit.

Please double-check me and make sure this is the right approach before we merge in this branch.

**Comment by HumbleBeeBumbleBee** [comment]
I guess the added size of SLS in application size and RAM usage is just the added external libraries.

Could you try caching songs in offline mode with your phone to see what is being cached?

I am having some issues with VLC, Spotify, Rocket Player, Stock Samsung Player, etc.
Most of these applications are hit and miss for scrobbling.

SLS seems to crashe sometimes when it tries to scrobble a song.
Google Play and the Stock Samsung Player are the only 2 applications I could get scrobbling.

**Comment by bryansills** [comment]
I'll look into that either tonight or over the weekend.

**Comment by bryansills** [comment]
Ok. I have some issues with VLC, but Soundcloud and Spotify work fine. However, I think the VLC issues are outside the scope of this pull request.

**Comment by HumbleBeeBumbleBee** [comment]
@bryansills I agree about VLC but I need to stack trace the crashes that I am having.
Any chance you could explain how I do that with my phone?
I have searched the internet, but I'm still not sure.
Any help is much appreciated.
Thanks!

**Comment by bryansills** [comment]
[Pidcat](https://github.com/JakeWharton/pidcat) is a good solution if you are running OSX. Otherwise, just use Android Studio.

**Comment by HumbleBeeBumbleBee** [comment]
Thanks!

**Pull request merged**